app/models/insight.rb

```diff
@@ -0,0 +1,63 @@
+# == Schema Information
+# Schema version: 20241024140606
+#
+# Table name: insights
+#
+#  id              :bigint           not null, primary key
+#  info_request_id :bigint
+#  model           :string
+#  temperature     :decimal(8, 2)
+#  template        :text
+#  output          :jsonb
+#  created_at      :datetime         not null
+#  updated_at      :datetime         not null
+#
+class Insight < ApplicationRecord
+  admin_columns exclude: [:template, :output],
+                include: [:duration, :prompt, :response]
+
+  after_commit :queue, on: :create
+
+  belongs_to :info_request, optional: false
+  has_many :outgoing_messages, through: :info_request
+
+  serialize :output, type: Hash, coder: JSON, default: {}
+
+  validates :model, presence: true
+  validates :temperature, presence: true
+  validates :template, presence: true
+
+  def self.client
+    @client ||= Ollama.new(
+      credentials: { address: ENV['OLLAMA_URL'] },
+      options: { server_sent_events: true }
+    )
+  end
+
+  def self.models
+    client.tags.first['models'].map { _1['model'] }.sort
+  end
```

Comment:
This could error when `OLLAMA_URL` isn't accessible from the web host. As it stands on myScoiety infra `OLLAMA_URL` is only available for the secondary sidekiq instance running on our dev server.

Reply:
I have changed the model select field to a text field, easiest solution to this for now but admins will need to know which models are pulled into Ollama.